doc/plugin_server_upstreamca_disk.md

```diff
@@ -17,9 +17,9 @@ The plugin accepts the following configuration options:
 | Configuration   | Description                                          |
 | ----------------| ---------------------------------------------------- |
 | ttl             | The TTL for issued certificates (deprecated)         |
-| cert_file_path  | Path to the "upstream" CA certificate                |
-| key_file_path   | Path to the "upstream" CA key file                   |
-| bundle_file_path| Path to the trust bundle that signed the upstream CA |
+| cert_file_path  | Path to the "upstream" CA certificate that corresponds to the signing key. |
+| key_file_path   | Path to the "upstream" CA key file. Key files must contain a single PEM encoded key. The supported key types are EC (ASN.1 or PKCS8 encoded) or RSA (PKCS1 or PKCS8 encoded).|
+| bundle_file_path| If the certificate specified above is not a root certificate, but has been signed by a different root authority, then it is also necessary to specify the certificate bundle so that the full trust chain can be verified. The certificate bundle file should contain a chain of certificates, in PEM format, up to the trusted root. If the certificate specified above is a root (that is, self-signed) certificate, then this does not need to be configured. |
```

Comment:
This isn't quite right. If self-signed, the file at `cert_file_path` should contain a single PEM encoded certificate representing the upstream CA certificate and `bundle_file_path` can be left unset. If not self-signed, then the file at `bundle_file_path` should contain one or more certificates representing the upstream "roots" and the file at `cert_file_path` contains one or more certificates necessary to chain up the the roots in `bundle_file_path` (where the first certificate in `cert_file_path` is the upstream CA certificate).

Reply:
See (https://github.com/spiffe/spire/blob/9b3c6f59afb27680e137983390a2b954683751e8/pkg/server/plugin/upstreamca/disk/disk.go#L155) for more context.